### PR TITLE
Update form for 2021 with new design, updated gender options, updated autocomplete, and more.

### PR DESCRIPTION
**Pull request by katrinafyi**
Relevant PRs: 
- #42 Implement new autocomplete with support for multiple majors.
- #43 Upgrade Stripe checkout, improve error handling, add script, and fix bugs.
- #44 Implemented more gender options and updated schema.

### CODE REVIEW

**Comment by katrinafyi** [comment]
image
image
image
image
image

**Comment by katrinafyi** [comment]
PTAL at new design. Migrated site to Bulma using the new customised build. Bulma does not support Bootstrap's toggle buttons so I've changed them to radios and dropdowns. 

At the moment, only the design is changed. Still lots to do on the backend and with Stripe...

File: uqcs/static/site.webmanifest [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "",
+    "short_name": "",
```
Comment (nicklambourne) [inline uqcs/static/site.webmanifest:3]: Would be nice to fill these at some point, though far from necessary in the short term, also the theme colours.

File: variables.sh.template [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
```
Comment (nicklambourne) [inline variables.sh.template:2]: Where does this file get used? Is this for deployment?

**Review by nicklambourne — APPROVED** [review]
Looks fantastic! A couple of minor comments/questions included.

My only real request for changes is that the dropdowns are made to be full width like the other input elements.

🎉

File: variables.sh.template (hunk 2 above)
Comment (JamesDearlove) [reply variables.sh.template:2]: If I was to guess, this would be for dev testing locally

**Review by JamesDearlove — APPROVED** [review]
Looks good, one improvement we could make though:

Now we have a static site, would it be worth moving the images and favicons to that site instead?

**Comment by katrinafyi** [comment]
Dropdowns have been widened. 

Re hosting images and favicons on static, it's probably not the best idea because they're only useful on the uqcs.org website. It makes more sense to have them here, I think.

**Comment by katrinafyi** [comment]
I'm going to keep working on this but submit PRs into this dev-2021 branch, so we can make all the relevant changes at once. I want to avoid someone stumbling upon join.uqcs.org, then announcing that it's been updated before it's done.

**Review by nicklambourne — CHANGES_REQUESTED** [review]
Not a fan of your single letter variables or the inconsistent usage of ' vs " throughout the Python, but both of those issues were here before your overhaul and I don't expect you to fix them as part of this. Otherwise the code looks fine.

A couple of things I noticed while testing out what I could (without access to API keys etc):
There's no Bachelor of Science (at least that shows up in program search), for some reason, despite it existing in the programs.json and programs_with_majors.json files:

Screen Shot 2021-02-09 at 10 43 11 pm

**Comment by nicklambourne** [comment]
After selecting a major the styling (particularly the sizing) of the becomes smaller (out of sync with the active selection). Not sure if this was intentional, but maybe greying it out would be more appropriate:
Screen Shot 2021-02-09 at 10 42 37 pm

**Comment by nicklambourne** [comment]
Failures on card payment aren't reported, but if you submit again it seems that it wrote to the DB anyway, based on what I can see:
Screen Shot 2021-02-09 at 10 55 51 pm
Screen Shot 2021-02-09 at 10 56 59 pm

**Comment by nicklambourne** [comment]
Interesting button display on all but the widest formats for the admin screen: 

Screen Shot 2021-02-09 at 11 03 08 pm

Not essential, but may be worth bumping the margins up a little.

**Comment by nicklambourne** [comment]
The app also appears to log nothing to the console under success or failure for user action (creation), which would make debugging incredibly hard were anything to ever go wrong. This is probably also a holdover from the past, but more critical than style issues.

**Comment by katrinafyi** [comment]
Thanks for the thorough testing. I'll review it again tomorrow.

Another bug: tapping an autocomplete suggestion with the touch screen inserts both the incomplete text and the suggested text.

**Comment by katrinafyi** [comment]
S few more responses below.

Science isn't missing, it's just the autocomplete is truncated to the first 10. This does make it impossible to find science. I'll think about how to fix this (probably a smarter ranking algorithm than alphabetical).

Actual card failures (e.g. wrong card info) should be handled by stripe on the checkout page. That error is due to missing the plan ID variable. 

It does work slightly differently to the existing system. Currently, it will only commit online members to the database on successful payment. Now, it adds the members when clicking the button which leads to what you see. 

Not sure what the best behaviour here is. Possibly, if an email already exists but is unpaid it should inform the user and give an option to pay via Stripe again.

**Review by jdcaperon — COMMENTED** [review]
🔥  it all and start again on Firebase? /s (not totally /s)

I think if we treat an email as a public key, we can use that as a check to whether a user has registered before.

1. Page one -> Enter email (Perform checks on whether email exists)
2. Page two -> If email exists enquire about updating details/payment or if not exists ask for new details
3. Page three -> Summary page of details and whether a user is paid or unpaid allowing them to loop back to the payment method.

We should favour getting peoples details over getting payment, i.e. committing on button press is fine as long as we use validation to not create multiple people.

**Comment by katrinafyi** [comment]
> I think if we treat an email as a public key, we can use that as a check to whether a user has registered before.

Something like that sounds ideal but tricky to get right. We should avoid displaying people's personal information just by entering emails. For the same reason, I don't think updating is possible in a secure way.

In the short term, the easiest solution is to save to the DB on successful payments only, so they can try again by entering their details again.

**Comment by katrinafyi** [comment]
Everything should be finalised. Tested mailing list and receipt with both online and in-person payments. Will merge soon today or tomorrow.

**Comment by katrinafyi** [comment]
Self-Merge-Approved. Good luck.

**Pull request merged**